### PR TITLE
Task/14371 Developer Menu: SRS Options

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
- Extends the SRS Options in the Developer Menu. 
- Shows the timeBetweenSubmissionsInDays in the SRS consent from App Configuration

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14371

## Screenshots
Screenshot 2022-12-02 at 13 33 53

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Developer Menu/Helper/DMMenuItem.swift [hunk 1]
```diff
@@ -146,7 +146,7 @@ extension DMMenuItem {
 		case .notifications: return "Trigger Certificate notifications only"
 		case .boosterRules: return "Download Booster rules and trigger the notification"
 		case .crashApp: return "Crash the App to test crash reporting."
-		case .srs: return "Disable some of SRS pre-checks."
+		case .srs: return "Configure some SRS Behavior"
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Developer Menu/Helper/DMMenuItem.swift]:
```suggestion
		case .srs: return "Configure SRS Behavior"
```

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift [hunk 2]
```diff
@@ -61,7 +61,7 @@ final class MockTestStore: Store, PPAnalyticsData {
 	var mostRecentRiskCalculationConfiguration: RiskCalculationConfiguration?
 	var forceAPITokenAuthorization = false
 	var recentTraceLocationCheckedInto: DMRecentTraceLocationCheckedInto?
-	var isSrsPrechecksEnabled = false
+	var dmIsSRSPreChecksEnabled = false
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift]: Don't know if this naming is consistent with our approach

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift [hunk 3]
```diff
@@ -41,7 +56,7 @@ struct SRSConsentViewModel {
 					UIImage(imageLiteralResourceName: "SRS-Warn-Others-icon"),
 					text: .string(String(
 						format: AppStrings.SRSConsentScreen.instruction2,
-						"90" // to.do fetch the number from app config
+						String(timeBetweenSubmissionsInDays)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift:59]: ❤️

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift (hunk 2 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift]: @30mar what do you think?

File: src/xcode/ENA/ENA/Source/Developer Menu/DMSRSPrechecks/DMSRSOptionsViewModel.swift [hunk 4]
```diff
@@ -71,37 +78,46 @@ final class DMSRSOptionsViewModel {
...
+		"""
 	}
 	
-	private func srsOTPString() -> String {
-		store.otpTokenSrs?.token ?? "No OTP Token set yet"
+	private func apiTokenStaticText() -> String {
+	   """
+	   
+	   PPAC API Token SRS
+	   \(store.ppacApiTokenSrs?.token ?? "No API Token generated yet")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Developer Menu/DMSRSPrechecks/DMSRSOptionsViewModel.swift:99]: Can you please update the screenshot as well?

**Review by naveeddotio — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift]: I think isSrsPrechecksEnabled is fine as it was, we use flags for dev menus without adding the dm part.

Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift]: Good point, renamed it to `isSrsPrechecksEnabled`

**Review by puneetmahali1 — APPROVED** [review]

**Pull request merged**